engine/script/src/script_hash.cpp

```diff
@@ -150,51 +201,35 @@ namespace dmScript
 
         dmHashTable64<int>* instances = &context->m_HashInstances;
         int* refp = instances->Get(hash);
+
         if (refp)
         {
-            lua_rawgeti(L, LUA_REGISTRYINDEX, context->m_ContextTableRef);
-            // [-1] Context table
-            lua_rawgeti(L, -1, *refp);
-            // [-2] Context table
-            // [-1] hash
+            lua_rawgeti(L, LUA_REGISTRYINDEX, context->m_ContextWeakTableRef);
```

Comment:
Always, check weak table. Registry needed only to increase ref counter. `instances` needed to know if the object still alive.